### ISSUE TITLE
SAB Deactivated Codes- showing up in Staged Fee List

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#535]
In GitLab by @kdjstudios on Oct 19, 2017, 13:38

**Submitted by:** "Cori Bartlett" <cori.bartlett@answernet.com>

**Helpdesk:** http://www.servicedesk.answernet.com/profiles/ticket/2017-10-19-82472/conversation

**Server:** Internal

**Client/Site:** Billings

**Account:** All

**Issue:** 

We deactivated a bunch of codes from the Billings site SAB. These codes are all now showing up in the Staged Fee list. (They previously weren’t.) Since they are deactivated, we don’t want to offer them as an option at all. Thanks! Cori

Test Cases: https://1drv.ms/x/s!AhaML4TTOKhybR02susmQUQwbfQ

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 19, 2017, 13:39

@smasih 

I can confirm that they have just deactivated 100+ codes on the Billings site, yet they still appear in the staged fee list.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Oct 24, 2017, 04:52

Hi [Kyle] ( @kdjstudios ),   
We have done the changes on [merge] (merge.sabilling.com) so that only Activated codes  will appear on the options list. You can check & if seems correct then please approve. We will release this on upcoming milestone.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 24, 2017, 08:19

@pchaudhary 

It was taking over 5 Minutes to load the Billings edit site page on Merge, so I tested using the Birmingham site. 

* When deactivating activities two confirmations are being displayed. I believe we only need one. Also when saving the site, it prompted "are you sure you want to reactivate this account?" Don't know why that is being displayed.

It does appear that deactivated fields are no longer showing in the staged fees dropdown on accounts.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Oct 25, 2017, 06:49

@kdjstudios @kace.coady @tim.traylor 

* It was taking over 5 Minutes to load the Billings edit site page
  * We will try to optimize site edit page on merge as soon as the current high priority tickets are completed.
*  When deactivating activities two confirmations are being displayed.
  * Yes, and both serve different purpose 
    * First one warns for activity activation/ deactivation
    * Second one warns that an email will be sent to the support or site manager on activation/ deactivation.
* Also when saving the site, it prompted "are you sure you want to reactivate this account?"
  * This is resolved and was only being displayed on merge.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 25, 2017, 08:21

@smasih 

In regards to the load time was this not already addressed with #199 ? 

Then if each serve a purpose, can we consolidate then into one prompt stating both that you are activating/deactivating and that an email will be sent. Having two prompts seems redundant, how do you feel @kace.coady @tim.traylor ?

Thank you for resolving the incorrect prompt when saving a site.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Oct 25, 2017, 09:26

@kdjstudios 

* In regards to the load time was this not already addressed with #199 ?
  * I am afraid but this ticket was never completed.
* Then if each serve a purpose, can we consolidate then into one prompt stating both that you are activating/deactivating and that an email will be sent. Having two prompts seems redundant, how do you feel @kace.coady @tim.traylor ?
  * Sure will do the changes.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 26, 2017, 08:21

@smasih @kace.coady @tim.traylor 

It appears that the prompts are already reduntant. As the second propmt already restates that first prompt. Please do Remove the first prompt and leave only the second prompt.

First Prompt:
![Untitled](/uploads/39a9c58a4b74cac6449a73f44ef7e6e6/Untitled.png)

Second Prompt:
![Untitled1](/uploads/03e648283fbba60f4a73e52195ed997d/Untitled1.png)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 26, 2017, 08:22

@smasih 

Will mark for approval once the prompts are corrected.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Oct 30, 2017, 04:45

moving to next milestone

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Nov 7, 2017, 04:18

Hi [Kyle] ( @kdjstudios ),   
As suggested, we have corrected the redundancy of prompts & create only a single prompt for the alert. The changes are live on production.  
Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 7, 2017, 13:24

@pchaudhary 

When I tried to deactivate an activity on MERGE within the Testy Site it did only show one prompt, however, it now also resulted in the following error:

![Untitled](/uploads/f8e5f8bac1f1374e6617f53948f991cb/Untitled.png)

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Nov 7, 2017, 22:37

Hi [Kyle] ( @kdjstudios ),   
Thanks for the screenshot, but I am not able to regenerate the error.  
Although I checked with activities, additional fees &  late charges & all went smoothly.  
Can you please try again?  
Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 8, 2017, 08:33

@pchaudhary 

I just retested and all is working now.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 10, 2017, 11:42

Marking for close.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 10, 2017, 11:44

Oops. I meant for production. Please release this on the 11-11 release.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Nov 10, 2017, 23:51

Hi [Kyle] ( @kdjstudios ),   
Thanks for the approval. We have released this on answernet production.   
Please check and update the label as well.  
Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 13, 2017, 08:39

Reached out to Cori for Final approval. Will post update once received.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 14, 2017, 07:53

@pchaudhary @tim.traylor @kace.coady 

Cori's Update:

Thank you! Looks good. I do have 1 other issue with the deactivated codes. Once the codes are deactivated, if I run a report that uses data from that code, it no longer shows up. Example: Revenue per Minute report. If one of the old codes for "Answering Service base rate" was deactivated, the revenue associated with that code no longer shows up on past reports. We need the codes deactivated so that they can't be selected as an option to assign to an account, but we still need to be able to pull reports on these. Cori

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Nov 14, 2017, 23:25

@kdjstudios 

I know Cori was descriptive enough but still, we are not able to understand the issue. Could you please explain it with example.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 15, 2017, 08:15

@smasih 

When you choose to run a report in a time range when the deactivated code was active and used; it does not appear in the report. 

In summary historical reports no longer work; due to deactivated codes no longer appearing in them.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Nov 15, 2017, 23:35

@kdjstudios 

ok, got it thanks for the explanation. We will look into it.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Nov 22, 2017, 24:56

Hi [Kyle] ( @kdjstudios ),   
We got the point what Cori's trying to say but still, we are unable to see any error.  
It'll be really helpful if Cori can provide us specific details with screenshots or something else through which we would be able to understand perfectly the issue.  
Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 22, 2017, 07:59

@pchaudhary @smasih @tim.traylor @kace.coady 

There is not an error (anymore). Cori brought up a second concern once that was resolved. Please read the last few posts. Simply the disabled codes are not showing on reports that are ran using a time range before the code was disabled. 

Marking for discussion

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 22, 2017, 12:28

Per @tim.traylor ; updating priority. This needs to be addressed immediately.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Nov 23, 2017, 08:21

@kdjstudios 

It would be really helpful if someone can give me the report name.
As per previous comments,  one report that got mentioned was Revenue per Minute Report... which only has account revenue details.

As per my understanding goes **Service Code Breakdown/Summary** is the report to figure out site total bifurcated on the basis of codes.

`071 800# + VM MV8` **is a deactivated code currently and not reflecting on the recent reports.**  
![Screenshot-3](/uploads/e314e864f4e2ed0f448dff5e0cbcda5b/Screenshot-3.png)  

**But if we pull `5/25/2016` we can see this code.**  
 ![Screenshot-2](/uploads/0c5350676dcf765ae4504ba730bf871c/Screenshot-2.png)  

Which proves that deactivated codes do reflect on historic data when they were active.

**Maybe this wasn't the report, please do let me know the _REPORT and CODES_ Cori is talking about. This info will really help us in expediting the process.**

**Comment by kdjstudios** [comment]
In GitLab by @abakshi on Nov 24, 2017, 06:15

The report is Service Code Breakdown and Side-by-Side report. 

The code should be available for reporting purpose even when they are disabled. 

@smasih - Need to check and share the findings on Side-by-Side report by following the similar approach & snapshots.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 24, 2017, 08:11

@tim.traylor @kace.coady @smasih @abakshi 

These are reports that I would think use activities in their calculations. Are there any others you can think of?

* Human Readable
* Revenue Per Minute
* Service Code Breakdown
* Side by Side
* Charges and Credits
* Billing Cycle Detailed
* Pending Staged Fees
* Revenue Analysis Worksheet
* Tax Report

In summary, any report that would use any activities in their calculations, right?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 24, 2017, 10:55

Update:

@tim.traylor and I discussed this further and found out that there are two possible concerns going on:


*  When Operations deactivated activities/additional fees they unchecked the boxes for Recurring, Taxable, and RPM; which would cause them to no longer be calculated.


* In the revenue per minute section of the site edit page we now will need to allow for selecting multiple activities to be included, rather then just one on the drop downs. Below is an initial draft of the functionality and layout we are discussing for the improved functionality. 

![RPM](/uploads/3a83c9a978fe02b27f90ff6a822fd826/RPM.png)

**Comment by kdjstudios** [comment]
In GitLab by @abakshi on Nov 27, 2017, 06:35

@tim.traylor provided the clarification, while @smasih will investigate into all the report that are using RPM codes except Revenue per Minute report.

**Comment by kdjstudios** [comment]
In GitLab by @abakshi on Nov 27, 2017, 06:35

changed time estimate to 30m

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Dec 26, 2017, 05:47

Moving to next milestone.

**Comment by kdjstudios** [comment]
In GitLab by @abakshi on Jan 4, 2018, 06:10

Will require to move this into next milestone. Sumeet will spend some time for ground work and once he is ready, this require a discussion with @tim.traylor

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 30, 2018, 13:37

Placing back into the ForDevelopment queue.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on May 26, 2018, 05:35

@kdjstudios @tim.traylor 

can we re-prioritize this ticket

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 29, 2018, 11:57

Please put this on the list for this Thursday to discuss.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 21, 2018, 09:02

@smasih @pchaudhary 

Any updates on this?

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Sep 18, 2018, 04:04

can we re-prioritize this ticket ?

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Sep 20, 2018, 05:29

Hi [Kyle] ( @kdjstudios ),   

We're sorry but we are unable to replicate the issue on Merge or Answernet. In order to resolve the same, we need to know the exact issue.

We would highly appreciate if you can provide us with a few more details with an example.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 20, 2018, 08:52

@pchaudhary @tim.traylor 

What issue were you testing? Please elaborate and how did you test it?

As what I can understand from reading the full ticket:
- Original issue reported is that deactivated activites, fees, were still able to be used as staged fees, etc. 
 - The fix was developed at http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/535#note_10533
- During testing of the above we found there are redundant prompts at http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/535#note_10676
 - the fix was developed at http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/535#note_11242 and was released http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/535#note_11343.
- After release Cori found an sub issue that the deactivated codes were no longer part of the reports http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/535#note_11473
 - Here is the list of reports we believe to be effected by deactivated codes: http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/535#note_11802
 - Here is the mock up of how to update the RPM site settings to account for deactivated codes. http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/535#note_11807

Nothing appears to have been done on this ticket for over the last 8-9 months now. So basicly we are needing to fix the last two items:

1. Insure reports are still correctly displaying deactivated codes, etc. in historical reporting. 
2. Update the RPM settings at the site level to insure the RPM report can do just that as well.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jan 24, 2019, 06:57

@kdjstudios 

Thanks for the update.  
We went through with the above comments & want to inform you that, we got the point what was shared by Cori.   
But it would be really helpful if you can provide a screenshot & other things to make it more clear on our end as we did not find a way to reiterate this.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jan 24, 2019, 06:57

removed milestone

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 24, 2019, 11:13

Please review with @tim.traylor in your morning meetings.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jan 28, 2019, 06:01

@kdjstudios 

We understand your point but for the discussion, we need some points or the current behavior of the scenario in which we have to discuss.   
For that purpose, we need an example at least to go further, we request you to please provide further details to discuss on.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 5, 2019, 09:34

- Insure reports are still correctly displaying deactivated codes, etc. in historical reporting.

> I think the above has already been addressed in following other tickets. 

- Update the RPM settings at the site level to insure the RPM report can do just that as well.

> We have a code **1000** that is a talk minutes charge. This code is selected on the **Total Minutes Activities** RPM settings at the site settings. We have billed using this code for the past three years. Now we are implementing a code **2000** that is a new Talk Minutes charge. We need to be able to select the **2000** code as an additional Total Minutes activities code, and we need to be able to keep the code **1000** for historical purposes. Additionally the code **1000** must continue to show on the historical RPM reports; while the new **2000** would only show up on the newer reports.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Feb 15, 2019, 05:44

Hi [Kyle] ( @kdjstudios ),   

Thanks for the updates & information & keeps eye on this ticket.   
We looked into this issue today & found the thing for which you're trying to explain. Please find below our findings.  


> Insure reports are still correctly displaying deactivated codes, etc. in historical reporting.

* No, the historical report is not showing the correct data, but the reason behind this is different than the deactivation of those fees (*additional/activities*).  
    * The deactivation of fees does not impact on the historical data but the uncheck of RPM affects the calculation.  
    * Which means if the operations deactivate it without unchecking the RPM check, then the reports will show the correct data.  

* As per my best understanding, the historical report data can be shown correctly by checking RPM checkbox true for deactivated fees as well.

* Now, the only thing to concern is to mandate the RPM check validation while deactivating such fees. Need to discuss with @tim.traylor 

> We have a code **1000** that is a talk minutes charge. This code is selected on the **Total Minutes Activities** RPM settings at the site settings. We have billed using this code for the past three years. Now we are implementing a code **2000** that is a new Talk Minutes charge. We need to be able to select the **2000** code as an additional Total Minutes activities code, and we need to be able to keep the code **1000** for historical purposes. Additionally, the code **1000** must continue to show on the historical RPM reports; while the new **2000** would only show up on the newer reports.

* Estimate -

|Size|Lower Time Bound|Variability|Upper Time Bound|Testing|Time for Testing| Time for Dev & Testing|
|-|-|-|-|-|-|-|
|Medium-1|4|Worked with Concepts, implementation ok|6|Low Impact|1.5|7.5|



Thanks  
Hope this helps.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Mar 28, 2019, 08:57

@pchaudhary Approved.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Mar 29, 2019, 05:58

@kdjstudios Kyle, can you pls do wording for a confirmation popup that warns Ops that unchecking the 'RPM' box for an activity can break historical reporting?  We'll add this feature in to help them not mess up the reporting.  Thx

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 29, 2019, 11:37

"By disabling the RPM for this activity may effect historical reporting of this activity or fee. Please click OK to confirm."

@tim.traylor @pchaudhary 

If they disable and then re-enable will that not break it? Or once its been disabled is it just broke?

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Mar 30, 2019, 04:49

Hi [Kyle] ( @kdjstudios ),  

Thanks for the verbiage.

> If they disable and then re-enable will that not break it? Or once it's been disabled is it just broke?

* No, it'll not break it, but only the historical report will not show that data.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Apr 2, 2019, 01:13

this is an enhancement as well

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Apr 2, 2019, 04:26

@kdjstudios 

* We have completed the development for selecting multiple **total calls** and **total minutes** for RPM under site page.

* We have updated the RPM report also.
* also please update the test cases(mention in the top of the Issue).

We have one question.
* Can we select same name/code multiple times for calls and minutes?


thanks

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Apr 2, 2019, 04:32

@kdjstudios 

Estimate for RPM Disabling: as per comment http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/535#note_37419 and http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/535#note_37448

|Size|Lower Time Bound|Variability|Upper Time Bound|Testing|Time for Testing| Time for Dev & Testing|
|-|-|-|-|-|-|-|
|R-Small|2|Concepts and Implementation Well Understood|2.5|Low Impact|0.5|3|

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Apr 2, 2019, 06:29

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 2, 2019, 08:38

Marking new estimate for Tim's Approval.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Apr 2, 2019, 09:41

@smasih Sumeet - I'm sure it won't be 3 hours for a popup confirmation, but I approve knocking this out.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Apr 3, 2019, 02:29

@kdjstudios 

Thank you for your approval.

We have complete the development and moving in our testing. we have mentioned in the spreadsheet above test cases and update to the test cases whatever you have.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 3, 2019, 10:34

@amishra @smasih @pchaudhary 

The link with the test cases only has one line? For some reason I would expect there are more then that. Please be sure you update the spreadsheet with all test cases needed to be completed. If it is only the one then please let me know. 

![image](/uploads/e5c85b4f5a40902d5561ae26d1fffb9c/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Apr 4, 2019, 24:17

@kdjstudios 

Thank you for the update.

As per my understanding, we have only two test cases for that. If you have another test cases then please update in the spreadsheet.

thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 4, 2019, 09:29

@amishra 

Thank you for the clarification and update. Please let us know when this is ready for our testing.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Apr 10, 2019, 05:59

@kdjstudios 

Thank you for the update.

We have setup branch on the dev server for your testing. Please test it.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 10, 2019, 10:23

@amishra 

This is not working. I just testing this on the Billings site and was not able to select deactivated codes to include. 

Drop down in the RPM settings at the site level. Not showing all codes.

![image](/uploads/fb3f68cc8746226d7802b245d6c989e8/image.png)

I should be able to see all codes that are marked with the RPM check box, regardless of active or deactivated. 

![image](/uploads/ba9b34d06640fa5fdde115d4e29329ee/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Apr 10, 2019, 23:07

assigned to @amishra and unassigned @pchaudhary

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Apr 11, 2019, 02:17

@kdjstudios 

Thank you for the update.


> I just tested this on the Billings site and was not able to select deactivated codes to include.

* Before starting the development,  we would like to mention here that the previous code was running in the same way. i.e Only shows active activities. doesn't matter RPM is true or false.

If you need to override the previous functionality then let us know we will change accordingly.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 11, 2019, 11:49

@amishra 

As pointed out in the http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/535#note_11794 yes we need to have the deactivated codes available for reporting in this section.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Apr 12, 2019, 05:22

@kdjstudios 

Thank you for the update.
 



> As pointed out in the http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/535#note_11794 yes we need to have the deactivated codes available for reporting in this section.

Yes, this will be available for the report section doesn't matter activities deactivated/activated.

### Before functionality.

**Inside the Site edit page.**
  * For RPM activity select box only selected those activities which are activated.  

**Inside the report section**
  * Inside the report he checked those activities/additional fee which marked for RPM true. (regardless of active or deactivated.)

### After develop the functionality

We have developed the same as previous functionality as mentioned above 2 points.

however, you need to change the functionality please let us know.


**Note: you will be able to see all codes that are marked with the RPM checkbox on the reports, regardless of active or deactivated**  

thanks.

Hope this help.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 12, 2019, 09:44

@amishra @smasih @pchaudhary @tim.traylor 

I am confused with why this is being marked for discussion. As pointed out in http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/535#note_38160 and agian in http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/535#note_38217 we need to have the deactivated billing codes that are RPM showing in the drop down section in the RPM section. As you can clearly see in the screen shots; none of the deactivated '071' activities that have RPM checkbox checked are showing the drop down list int he RPM section.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Apr 13, 2019, 01:04

@kdjstudios 

Thank you for the update.



> I should be able to see all codes that are marked with the RPM check box, regardless of active or deactivated.

We will change the RPM functionality accordingly. We will let you know when ready for testing.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Apr 19, 2019, 01:56

@kdjstudios 

We have completed the development and we have set up a branch on the merge server. 
please test it.


Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 19, 2019, 08:12

@amishra 

Thank you for the update. This is working as expected. I was able to add more then one RPM item in the RPM settings at the site and remove them as will without issue. It allowed for using deactivated codes in the drop down too. I ran the RPM report for an older billing cycle for the Billings site and was able to have it match the usage for that billing cycle. This appears to be working correctly. 

Marking for announcement.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Apr 27, 2019, 05:29

Hi [Kyle] ( @kdjstudios ), 

We have successfully released this on both of the production servers.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 27, 2019, 12:09

I was able to test and confirm on the demo site on production that I could add more then one fee at the stie edit page and was also able to successfully run the RPM report.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 30, 2019, 14:47

I wrote:

> We are just following up. After review it was decided to at the site level allow for multiple codes to be selected for RPM settings. This includes allowing to select deactivated codes in here to ensure historical reporting is not effected. The report itself was then updated to use all of the selected codes in the RPM section. This update has been released onto production. Please let us know if there are any questions or concerns.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 1, 2019, 11:51

closed
